### ISSUE TITLE
Control pipelines: crash when converting empty reading to operation

### ISSUE THREAD

**Issue by FlorentP42** [fledge-iot/fledge-service-dispatcher#52]
**Describe the bug**
When exiting a control pipeline, if the reading does not contain any value, the dispatcher will crash with the following stack trace:
```
Fledge dispatcher[1741]: FATAL: Signal 11 (Segmentation fault) trapped:
Fledge dispatcher[1741]: FATAL: (0) 0       0x5628b82807d2 handler(int) + 82---------
Fledge dispatcher[1741]: FATAL: (1) 1       0x7fcbb5b31090 /lib/x86_64-linux-gnu/libc.so.6(+0x43090) [0x7fcbb5b31090]---------
Fledge dispatcher[1741]: FATAL: (2) 2       0x5628b82de657 PipelineExecutionContext::filter(Reading*) + 311---------
Fledge dispatcher[1741]: FATAL: (3) 3       0x5628b827f5da ControlOperationRequest::filter(ControlPipelineManager*) + 202---------
Fledge dispatcher[1741]: FATAL: (4) 4       0x5628b827f841 ControlOperationServiceRequest::execute(DispatcherService*) + 65---------
Fledge dispatcher[1741]: FATAL: (5) 5       0x5628b82bb623 DispatcherService::worker() + 35---------
Fledge dispatcher[1741]: FATAL: (6) 6       0x7fcbb5dd3df4 /lib/x86_64-linux-gnu/libstdc++.so.6(+0xd6df4) [0x7fcbb5dd3df4]---------
Fledge dispatcher[1741]: FATAL: (7) 7       0x7fcbb634c609 /lib/x86_64-linux-gnu/libpthread.so.0(+0x8609) [0x7fcbb634c609]---------
Fledge dispatcher[1741]: FATAL: (8) 8       0x7fcbb5c0d133 clone + 67---------
```

**To Reproduce**
1. Create a north service that can call an operation "TestOP" containing one parameter "type".
3. Create a south service that can receive  an operation "TestOP" with 0 parameters that prints a log when receiving an operation.
4. Create a control pipeline between those services.
5. Add a filter to this pipeline that will remove the paramters from any operation it process and log the resulting reading.
6. Restart Fledge so that the pipeline becomes active.
7. Call the "TestOP" operation from the north plugin.
8. In the log from the filter, notice that the reading content is properly removed:  `{"asset_code":"TestOP","user_ts":"2023-12-04 14:38:07.746647+00:00","ts":"2023-12-04 14:38:07.746647+00:00","reading":{}}`
9. Notice the crash before the operation could reach the south plugin.

**Expected behavior**
The reading is properly converted into an operation with 0 parameters.

**Additional context**
N/A

**Comment by FlorentP42** [comment]
I encountered this on 2.2.0, but I can't seem to reproduce it on develop, was this maybe fixed by some commit in the meantime?

**Comment by ashish-jabble** [comment]
@FlorentP42 yes this case has been handled in develop branch. Here is related [Pull Request](https://github.com/fledge-iot/fledge-service-dispatcher/pull/62/files).

**Comment by aklira** [comment]
Test OK, issue can closed

**Comment by FlorentP42** [comment]
Reopening as this issue have reappeared in Fledge 2.3.0-47 nightly:
```
Fledge dispatcher[1361]: DEBUG: hnz_pivot_filter - PivotCommand - HNZPivotFilter::ingest - converted Reading: {"asset_code":"PivotCommand","user_ts":"2024-01-26 10:55:48.404169+00:00","ts":"2024-01-26 10:55:48.404169+00:00","reading":{}} 
Fledge dispatcher[1361]: INFO: Control filter pipeline iec104north_c1_to_hnzsouth_s1 removed control request
Fledge dispatcher[1361]: FATAL: Signal 11 (Segmentation fault) trapped:
Fledge dispatcher[1361]: FATAL: (0) 0 0x564002487dc2 handler(int) + 82---------
Fledge dispatcher[1361]: FATAL: (1) 1 0x7fd51cffd090 /lib/x86_64-linux-gnu/libc.so.6(+0x43090) [0x7fd51cffd090]---------
Fledge dispatcher[1361]: FATAL: (2) 2 0x5640024e34e9 KVList::fromReading(Reading*) + 121---------
Fledge dispatcher[1361]: FATAL: (3) 3 0x564002486b0e ControlOperationRequest::filter(ControlPipelineManager*) + 190---------
Fledge dispatcher[1361]: FATAL: (4) 4 0x564002486e41 ControlOperationServiceRequest::execute(DispatcherService*) + 65---------
Fledge dispatcher[1361]: FATAL: (5) 5 0x5640024c2913 DispatcherService::worker() + 35---------
Fledge dispatcher[1361]: FATAL: (6) 6 0x7fd51d29fdf4 /lib/x86_64-linux-gnu/libstdc++.so.6(+0xd6df4) [0x7fd51d29fdf4]---------
Fledge dispatcher[1361]: FATAL: (7) 7 0x7fd51d819609 /lib/x86_64-linux-gnu/libpthread.so.0(+0x8609) [0x7fd51d819609]---------
Fledge dispatcher[1361]: FATAL: (8) 8 0x7fd51d0d9353 clone + 67---------
```

**Comment by FlorentP42** [comment]
One more test with Fledge 2.3.0-95 / dispatcher 2.3.0-13 still produces the crash:
```
Fledge dispatcher[5573]: DEBUG: hnz_pivot_filter - PivotCommand - HNZPivotFilter::ingest - converted Reading: {"asset_code":"PivotCommand","user_ts":"2024-02-12 11:49:09.831791+00:00","ts":"2024-02-12 11:49:09.831791+00:00","reading":{}} 
Fledge dispatcher[5573]: INFO: Control filter pipeline iec104north_c1_to_hnzsouth_s1 removed control request
Fledge dispatcher[5573]: FATAL: Signal 11 (Segmentation fault) trapped:
Fledge dispatcher[5573]: FATAL: (0) 0 0x557a776b2dc2 handler(int) + 82---------
Fledge dispatcher[5573]: FATAL: (1) 1 0x7fe0f9dc8090 /lib/x86_64-linux-gnu/libc.so.6(+0x43090) [0x7fe0f9dc8090]---------
Fledge dispatcher[5573]: FATAL: (2) 2 0x557a7770e4e9 KVList::fromReading(Reading*) + 121---------
Fledge dispatcher[5573]: FATAL: (3) 3 0x557a776b1b0e ControlOperationRequest::filter(ControlPipelineManager*) + 190---------
Fledge dispatcher[5573]: FATAL: (4) 4 0x557a776b1e41 ControlOperationServiceRequest::execute(DispatcherService*) + 65---------
Fledge dispatcher[5573]: FATAL: (5) 5 0x557a776ed913 DispatcherService::worker() + 35---------
Fledge dispatcher[5573]: FATAL: (6) 6 0x7fe0fa06adf4 /lib/x86_64-linux-gnu/libstdc++.so.6(+0xd6df4) [0x7fe0fa06adf4]---------
Fledge dispatcher[5573]: FATAL: (7) 7 0x7fe0fa5e4609 /lib/x86_64-linux-gnu/libpthread.so.0(+0x8609) [0x7fe0fa5e4609]---------
Fledge dispatcher[5573]: FATAL: (8) 8 0x7fe0f9ea4353 clone + 67---------
```

**Comment by FlorentP42** [comment]
The issue appears to be fixed in Fledge 2.4.0 (please make sure all changed on this release are included in develop to avoid having to reopen this issue like last time).
